<launch>

  <param name="use_sim_time"  value="true"/>

  <arg name="bag_file" doc="path to the bag_file" />

  <!-- Broadcast missing TF -->
    <node pkg="tf" type="static_transform_publisher" name="tf_base"           args = "0 0 0 0 0 0 base_footprint base_link 50"/>
    <node pkg="tf" type="static_transform_publisher" name="tf_camera_link"    args = "0.17 0 0.25 0 0 0 base_link camera_link 50"/>
    <node pkg="tf" type="static_transform_publisher" name="tf_camera"         args = "0 0 0 0 0 0 camera_link camera 50"/>
    <node pkg="tf" type="static_transform_publisher" name="tf_camera_opt"     args = "0 0 0 -1.570796 0 -1.57 camera camera_color_optical_frame 50"/>
    <node pkg="tf" type="static_transform_publisher" name="tf_map"            args = "0 0 0 0 0 0 map odom 50"/>

  <!-- Launch localization -->
    <include file="$(find localization)/launch/localization.launch" />


  <!-- If use scan is true, launch the appropriate -->
  <!-- pointcloud converter and the pseudo scan node -->
    <!-- Pointcloud converter -->
    <node pkg="nodelet" type="nodelet" args="manager" name="depth_to_pointcloud_manager" output="screen" />
    <node pkg="nodelet" type="nodelet" name="reasense_depth2pointcloud" args="load depth_image_proc/point_cloud_xyz depth_to_pointcloud_manager" output="screen">
      <remap from="/image_rect"      to="/camera/aligned_depth_to_color/image_raw"   />
      <remap from="/camera_info"     to="/camera/aligned_depth_to_color/camera_info" />
      <remap from="points"           to="/new_pointcloud"/>
    </node>

    <!-- PCL filter -->
    <node pkg="sensors" name="pcl_filter" type="pcl_filter" output="screen" >
        <param name="robot_height"  value="0.4" />
        <param name="max_height"    value="1.2" />
        <param name="n_scans_above" value="1" />
        <param name="n_scans_below" value="1" />
        <remap from="pcl" to="new_pointcloud" />
    </node>

    <!--node pkg="sensors" name="min_scan"  type="min_2Dscan" output="screen">
      <remap from="pcl" to="new_pointcloud" />
    </node-->

    <!-- Obstacle perception -->
    <include file="$(find sensors)/launch/obstacle_tracker.launch" />

    <!-- Play rosbag -->
    <!-- As the tfs are not properly setup in some old bag files, tfs are not published -->
    <node pkg="rosbag" type="play" name="player" args=" --clock $(arg bag_file) --topics
        /cmd_vel /camera/color/image_raw
        /camera/aligned_depth_to_color/camera_info /camera/aligned_depth_to_color/image_raw
        /camera/infra1/image_rect_raw /camera/infra2/image_rect_raw /camera/infra1/camera_info"/>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find sensors)/rviz/offline_obstacles.rviz"/>

</launch>
